internal/provider/resource/secret_resource.go

```diff
@@ -692,3 +693,43 @@ func (r *secretResource) Delete(ctx context.Context, req resource.DeleteRequest,
 	}
 
 }
+
+func (r *secretResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+
+	secret, err := r.client.GetSingleSecretByIDV3(infisical.GetSingleSecretByIDV3Request{
+		ID: req.ID,
+	})
+
+	if err != nil {
+		if err == infisical.ErrNotFound {
+			resp.Diagnostics.AddError(
+				"Secret not found",
+				"The secret with the given ID was not found",
+			)
+		} else {
```

Comment:
looks like we're missing this 

	if response.StatusCode() == http.StatusNotFound {
		return {}, ErrNotFound
	}